test/Tests/src/Data/Vector_Spec.enso

```diff
@@ -25,6 +25,15 @@ compare_tco a b = case a.vec.length == b.vec.length of
 foreign js generate_js_array = """
     return [1, 2, 3, 4, 5]
 
+foreign js generate_nested_js_array = """
+    return [[1, 2, 3], [4, 5]]
+
+foreign python generate_py_array = """
+    return [1, 2, 3, 4, None]
+
+foreign python generate_nested_py_array = """
+    return [[1, 2, 3], [4, 5]]
```

Comment:
I think I'd also like to see a test where these two situations are joined - a nested array containing a null - so that we can correctly verify that not only surface level nulls are converted but also ones deep into the nesting.

Reply:
That's unlikely to work atm - see the pending test I added. It also demonstrates that it wouldn't work even before this PR.
In order to support it, we need a bigger rewrite of polyglot arrays which is not part of this PR.

Reply:
Oh, I missed that it is pending. That is concerning - is the fix for that planned for the near future?

Well, I'd suggest adding this mixed test as pending too - to ensure that, once we do the rewrite, stuff is handled correctly.

Reply:
> That is concerning

Yes, I also find this concerning.